mlflow/store/ftp_artifact_repo.py

```diff
@@ -30,69 +31,66 @@ def __init__(self, artifact_uri, client=None):
 
         super(FTPArtifactRepository, self).__init__(artifact_uri)
 
+    @contextmanager
     def get_ftp_client(self):
         if self.client:
-            return self.client
+            yield self.client
```

Comment:
Ah, I see we currently need to have this branch since we rely on being able to pass `ftp_mock` as the FTP client in our unit tests. The only issue with doing that is that we never actually test the `else` branch, which is the case we expect users to hit. Since we already don't exercise the user-facing logic in `get_ftp_client` in our test, maybe we could just mock `mlflow.store.ftp_artifact_repository.FTPArtifactRepository.get_ftp_client` in our tests instead of adding an extra param to FTPArtifactRepository that we only use for testing. However I think it's also ok to make this change in a follow-up PR, if you'd be interested in doing that. Let me know what you think!

Reply:
Thanks for suggestion, I will try to fix this.